contributors/design-proposals/arbitrator.md

```diff
@@ -0,0 +1,284 @@
+ "Compute Resource Quota" and "Storage Resource Quota"; it can not exceed resources in hard section.
+ If ResourceQuota Admission enabled, it’ll also check whether total “reserved” exceed resources in cluster.
+* hard: The hard section defines the max resources that a namespace can use; it can not exceed `Quota.hard` of namespace if any. 
+
+The `Consumer` is created by arbitrator for each namespace and updated by cluster admin if necessary; the arbitrator creates
+ the `Customer` with infinite `hard` and "zero" `reserved`, so the namespace shares cluster resource equally by default. 
+
+And the arbitrator will create or update `Allocation` with additional field `deserved`: 
+* deserved: Similar to “Used” of Quota, it’s not defined in yaml file but updated by arbitrator. It defines the total resources
+that arbitrator allocates to a namespace. It’s not more than hard of Quota, and maybe changed because of workload in namespaces.
+* hard/reserved: Copy from `Consumer`; if `Consumer` was updated, it's also updated in next arbitration cycle 
+
+```yaml
+apiVersion: kuabe-arbitrator.incubator.k8s.io/v1
+kind: Allocation
+metadata:
+  name: defaults
+spec:
+  deserved:
+    cpu: "1.5"
+  hard:
+    requests.cpu: "1"
+    requests.memory: 1Gi
+    limits.cpu: "2"
+    limits.memory: 2Gi
+  reserved:
+    limits.memory: 1Gi
+```
+
+The following figure demonstrate the relationship of hard, reserved and deserved in `Consumer`/`Allocation`.
+
+> NOTES: only __Compute Resource__ and __Storage Resource__ are available for reserved & deserved.
+
+```text
+    -------------  <-- hard
+    |           |
+    |           |
+    - - - - - - -  <-- deserved
+    |           |
+    |           |
+    |           |
+    -------------  <-- reserved
+    |           |
+    -------------
+```
+
+The `k8s-arbitrator` is a new binary that created/updated `Consumer` and `Allocation`; it'll
+
+1. Calculate deserved resource (`Allocation.deserved`) based on arbitrator’s policy, e.g. DRF, and
+  namespace’s request (pending pods in PoC)
+1. Evict Pods of overused namespace
+1. Provide predicates that “namespace will not exceed Allocation.deserved” as scheduler's HTTPExtender
+
+Meanwhile, `k8s-scheduler` dispatch tasks to host based on its policy, e.g. PodAffinity:
+`k8s-arbitrator` own resource allocation, `k8s-scheduler` own resource assignment.
+
+The arbitrator takes DRF as default policy. It will list pods/nodes from `k8s-apiserver`, and calculate 
+the deserved resource of each namespace based on DRF algorithm; and then, update the deserved to `Allocation` accordingly.
+The default interval is 1s which is configurable. The arbitrator will NOT assign hostname to the deserved resources;
+it dependent on scheduler to dispatch tasks on the suitable hosts.
+
+The arbitrator also meets the following requirements:
+
+* Total deserved resource of namespace can not exceed resources in cluster
```

Comment:
what happens if nodes die?  what happens if my nodes are tainted and could not match the pods being asked to schedule?  i struggle to see how we will enforce deserved resource <= cluster allocatable, and would argue that we should avoid this use case to some degree.